src/commontransformations.jl

```diff
@@ -310,56 +317,89 @@ Base.show(io::IO, r::RotationXY) = print(io, "RotationXY($(r.angle))")
 Base.show(io::IO, r::RotationYZ) = print(io, "RotationYZ($(r.angle))")
 Base.show(io::IO, r::RotationZX) = print(io, "RotationZX($(r.angle))")
 
+# It's a little fiddly to support all possible point container types, but this should do a good majority of them!
+transform(trans::RotationXY, x::Vector) =    [x[1]*trans.cos - x[2]*trans.sin, x[1]*trans.sin + x[2]*trans.cos, x[3]]
+transform(trans::RotationXY, x::NTuple{3}) = (x[1]*trans.cos - x[2]*trans.sin, x[1]*trans.sin + x[2]*trans.cos, x[3])
 function transform(trans::RotationXY, x::FixedVector{3})
     (sincos, x2) = promote(Vec(trans.sin, trans.cos), x)
     (typeof(x2))(x[1]*sincos[2] - x[2]*sincos[1], x[1]*sincos[1] + x[2]*sincos[2], x2[3])
 end
+function transform{T}(trans::RotationXY{T}, x)
+    Z = zero(T)
+    I = one(T)
+
+    [trans.cos -trans.sin Z;
+    trans.sin  trans.cos Z;
+    Z          Z         I ] * x
+end
```

Comment:
Could do with some extra tests if the implementation of these transformations is going to be repeated for each input type.  Also could have the tuple (for eg) as the definitive implementation and let the other versions rely on that, though that won't work for the generic version here.  (do we even have a type which would actually hit this generic version?)